lib/ShareManifests.js

```diff
@@ -0,0 +1,108 @@
+    "homepageURL": "https://twitter.com",
+    "icon32ULR": "images/twitter/icon32.png",
+    "icon64URL": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAEAAAABACAIAAAAlC+aJAAAFgElEQVR42u2Za2wUVRSApy1CFWvBYkARrMZgor9MJRgfkR/8MJEfxPjHYIBE0YTE9w9DYkIkYtQYY2I0QdFoCBV2Z3d2l+5SKC2FKg/7gLZItbS0hZY+5t47M7uz89yZ8dzdhqYJyI47UyWZk5NJNp3mnu+ce8695wzDxPAtrQFAABAABAABQAAQAAQAAcDcKoeZKGIiiD65WwgAbGURE+KZMF/JoYUxvICDn6A8wFznZUroIQB1WwnWw/+GUU2SbDiT/uSiuu+KHhnT9l7Rd/Wr637LlMcQ/JXhrplOX56fIJUNhOE8ApgXx1UNNPTuracGlbPoxdNSM28Ihu3MlgnN3jeqPXE8zYTyGCyqSZBNnfL3w2ptk0iDFi8RII4h6HVt0v5RbVWjwBzgXfoelbFo2zkZ67ZzYxmQcxs70s+dkLZfUNqQqebsb4fVymJ2UTGbpyyEdvSpsEzkqr48SairuGI3Hli/pTNDHX8zyeZsrFuF91KTxjJw3H6eSZQOEMGVcRy9qheWaZw0Hj4iQOYVxcCix5qFoWzOKVps22mZMh5tFuuahNe75RVNAhMtFQDdmSItvHFtjd+J8fRxiWFvFt8oBfhiQHHcSNqwI2P6zyPakGJ9PahUx7AHAFUpIQ8wI5cV662ebDVND8BANwrd3UncLpquAEzbKWyj+iv6ojh1gQdb6PYEgU3pzBbVcg5PGutPpmmqHZiu5bMqRohf0yqNKJbjUgDguyHtngYC3vHiHIjiChZ9c0m97mJIt3cPaevapKoERAOMRpSEmwZYfzrDQ/FxI7rlfHZRqaK+5z06iePUlBdOZaQbVxKiWw0Txvt/KE+2inAAlbG08jL7JteeEK+q7gDGNbv2iAi+9/QqwaJqDh+avMluBksvZa0W3twzpH3Qm91wSnqzWx7XXAM80AwVAnsKAEH4ZeqRJvGcVGxBNCxHNCxiWDnbcSVjKgCIs25HpSfxkkbyRo9c1yJu7syMQ1L6KefT1r1HBE8BWHRfo9Ah5rBhJ8eNgUzOVwA4KKuh/kQ93EIcvo1DUJWdOZHdI9o8uqi3ORDit3XLpu279bDC2+cVhvUcgEXLU6RdyPkNAFfr539NQw31oSMLoY0dspKz/U2AKWNJEsqGHwARXMHxO/9UND+L0K6/shBt33riKKqM4Vc6MqewiXTb8joYcAI8BZfcMO9jU18Rw4uTpK5V/GpARarHsYAWeX6Epq9vABG0uAH/OKJ1S7lOwRQNL0MwrlprqPuhA/YPgENlEfTqWdnxQb4cVMtpB+z3XCiKF8TIDyOat9afk6yHDhNI3zkZbIXRiqQQHtO9qqhYt186IxeabL8BZhhqEvjjfuWiXOrRZljOhxeUCjrVQnM7Wszv19Wt0s5+FQYWA5l/GY/6UWNRnBSmV3MIEJ8+Fujcpp5ffUw8iU3HvSQnDLjnzm4d3Wuxw2QOwZNqBBVmtPB8MEXe7cl2EBMaWbejhz3D2tJD0773F6A8DgwI/ARaFkXQbq9MkWfa0p/2K12iqbo/kHnd2tGXvSufS/5/H+DIsqPie73Zj/qV7X3Zzwe0+lG9W8xJpj3jdDdTt2O8ue6kVA6mRwpL+A6AFx4kL7dnYGHVKqXa2NBPv9Oj3J8iriqmN3MhuGDBkGzD6czey/pg1iq+5OiWPaFZMFWH8UQt5GuI5s9/8okpv/ABdEcUP35U2NyV+WlEO0PMQdmCsY9o2OnctMLWmtLs4azVK5kNEzrMyteekJYmaDtBNVqoYJ6q69IZpiWokuVrDpKVTcKzbdLG9vTWLvm1LnnrWXlThwwt1apmYWmKVHF5u8MzU8f/00c+jup0SWX5WQqEbD5HueAzawAQAAQAAUAAEAAEAAFAAPCP+jdTEm8kfrK2ewAAAABJRU5ErkJggg==",
+    "iconURL": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAIAAACQkWg2AAABEElEQVQoz2Ng2PCWNERVDevBCKcGNOl1b7k3vQMihjVvUMThqg33f5Dd8Z5h7RuI6rCTn298+gNEDde+RZ76zIyuYc2bumvfrnz8Y3/wI8Pat+wb3z769vc/DLTc+MayHk3DujeuRz7+/Pv/19//u17+mnzn+7c/UNWffv9T3PmeAV3D2jeRpz///fcfE9z/+pdvyzsMDevfCmx+t+PlL0wN8x78gHoMPZTWvsk8/+X7HxRbPvz6p7PnA7ZQ2gDyqNCmt723vsFVf/79Dxg+KMYjO8n64McVj3+8/vnvx99/L378Xfvkp8X+j+iq0SKOd/M71d3vdfZ+EN/2HhhuwNggImmsB8UalhRBm8SHDQEA5UPgBrwMoL4AAAAASUVORK5CYII=",
+    "name": "Twitter",
+    "origin": "https://twitter.com",
+    "pageSize": {
+      "share": {
+        "height": 300,
+        "width": 550
+      }
+    },
+    "shareURL": "https://twitter.com/intent/tweet?url=%{url}&text=%{text}",
+    "postActivationURL": "https://activations.cdn.mozilla.net/en-US/activated/twitter.html"
```

Comment:
Is `version` a required field in this schema? I noticed it was missing for Twitter.
Or more specifically/annoyingly, I wrote a quick Joi schema validation for these manifests and it complained about the following:

```
social.manifest.twitter-com:
  child "icon32URL" fails because ["icon32URL" is required].
  child "version" fails because ["version" is required].
  "icon32ULR" is not allowed.
```